sources/packages/backend/apps/api/src/route-controllers/audit/audit.controller.ts

```diff
@@ -0,0 +1,38 @@
+import { Controller, Post, Param, Req } from "@nestjs/common";
+import { ApiTags } from "@nestjs/swagger";
+import { AuthorizedParties, IUserToken } from "../../auth";
+import { AllowAuthorizedParty, UserToken } from "../../auth/decorators";
+import BaseController from "../BaseController";
+import { Request } from "express";
+import { AuditEvent, AuditService } from "../../services";
+
+@AllowAuthorizedParty(
+  AuthorizedParties.institution,
+  AuthorizedParties.student,
+  AuthorizedParties.supportingUsers,
+  AuthorizedParties.aest,
+)
+@Controller("audit")
+@ApiTags("audit")
+export class AuditController extends BaseController {
+  constructor(private readonly auditService: AuditService) {
+    super();
+  }
+
+  /**
+   * Audit events log.
+   * @param event audit event.
+   * @param userToken user token.
+   * @param request request object.
+   */
+  @Post(":event")
```

Comment:
Was there an advantage to using the `:event` as a URL parameter instead of a property in the POST body?

Reply:
I think it's simpler to implement and easier to understand what's being done this way.

Reply:
I do not agree with either argument.
In terms of expanding the implementation, a body would be better.
In terms of providing arguments to a POST, the body would be preferred.
We can ask another team member, I would not say that it is a blocker.
@dheepak-aot thoughts or preferences on this?
@sh16011993 please share your opinion as a reviewer.

Reply:
From the API perspective `event` is the payload itself here and not the context of the payload. Also with that in mind, making the event in body would make the payload extensible.
Hence my recommendation would be body.

Reply:
ok. Moved it to the body.